docs/contributing.rst

```diff
@@ -2,12 +2,575 @@
+example,
+
+* Write or revise documentation (including this document)
+* Implement data analysis operators for unstructured grids from scratch
+  or from their workflows
+* Develop example notebooks that demonstrate how a particular function
+  is used
+* Answer a support question
+* Simply request a feature or report a bug
+
+All of these activities are signicant contributions to the on-going
+development and maintenance of UXarray.
+
+1.3. About this guide
+---------------------
+
+The UXarray content is hosted on GitHub, and through this document,
+we aim to ease the community's experience with contributing to this project.
+However, this guide might still be missing case-specific details; please do
+not hesitate to reach out to us to consult any such cases.
+
+.. note::
+    Much of the information in this guide has been co-opted from the
+    `GeoCAT <https://geocat.ucar.edu/pages/contributing.html>`_ project and
+    `Project Pythia <https://projectpythia.org/contributing.html>`_.
+
+1.4. Project Specific Resources
+-------------------------------
+
+Some important UXarray resources are as follows:
+
+* `UXarray GitHub repository <https://github.com/UXARRAY/uxarray>`_ houses
+  the open-source code base along with some significant documentation such
+  as `Readme <https://github.com/UXARRAY/uxarray/blob/main/README.md>`_ and
+  `UXarray draft API <https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_.
+
+* `UXarray documentation <https://uxarray.readthedocs.io/>`_
+  houses significant documentation such as :ref:`quickstart`, :ref:`installation`,
+  :ref:`contributing` (i.e. this document), :ref:`examples`, :ref:`tutorials`,
+  and :ref:`api`.
+
+* `UXarray draft API <https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_
+  tentatively shows the eventual list of UXarray functionalities. It is
+  open to community feedback and inputs!
+
+  - Please create a
+    GitHub `Discussion <https://github.com/UXARRAY/uxarray/discussions>`_ or
+    `Issue <https://github.com/UXARRAY/uxarray/issues>`_ if you feel any changes
+    should be made to this document!
+
+2. Configuring GitHub and Git, and Setting Up Python Environment
+================================================================
+
+In this section, we provide details about how to configure Github and Git
+in order to work with the UXarray GitHub repository. This is because most
+contributions, such as adding a new function or making changes to an
+existing one, writing a usage example, or making some modifications to this
+Contributor's Guide requires operating directly on the GitHub repository.
+
+Contributing to a GitHub repository follows almost the same process by
+any open development Python project maintained on GitHub.
+However, it can still seem complex and somewhat varied from one project to
+another. As such, we will provide an overview of GitHub here and refer the
```

Comment:
I think we should assume contributors have basic git and github knowledge.

Reply:
I think this might be a reasonable assumption, but I've definitely had some users or potential contributors to other projects not have a github. We should keep this section in, I think, to err on the side of not assuming prior knowledge as much as possible.

Reply:
I do.  I really want developers to have already learned the basics of git/github on their own before contributing to uxarray.  The presences of this material implies we want total novices contributing and I don't think we do.

Reply:
@rljacob I can see where you're coming from, but a total novice to git/github might not be a novice with respect to python coding, algorithm development, or good ideas.

Reply:
I don't want to discourage git novices but I don't want UXarray to take on the responsibility for teaching them git/github.  We should just point them to our favorite websites for that and then focus on the UXarray-specific parts of contributing code.

Reply:
Ok, this makes sense! I have removed the details of GitHub/Git configuration and instead put some text that refers the refer to the comprehensive guides of those. Please let me know if the new look suffices for your concerns.